app/views/users/_user.html.erb

```diff
@@ -1,6 +1,16 @@
 <li>
-  Name: <%= user.name %>
-  <span class="profile-link">
-    <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
-  </span>
+    Name: <%= user.name %>
+    <span class="profile-link">
+      <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
+    </span>
+  <% if !Friendship.find_by(user_id: current_user.id, friend_id: user.id) and !Friendship.find_by(user_id: user.id, friend_id: current_user.id) and current_user != user %>
```

Comment:
There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.

Reply:
Timeline shows own posts too now, removed most of the logic from the views (your first example was part of the microverse-repo I forked, it wasn't written by me), added specs for likes, comments and posts with all kind of tests

Reply:
Also all linters should be installed now